security/pkg/pki/ca/ca.go

```diff
@@ -232,15 +260,49 @@ func NewSelfSignedDebugIstioCAOptions(rootCertFile string, caCertTTL, defaultCer
 	return caOpts, nil
 }
 
+func specifySelfSignedAlgorithmIstioCAOptions(opts *IstioCAOptions, caRSAKeySize int, ecSigAlg, eccCurve string) error {
+	switch opts.AlgorithmType {
+	case util.RsaAlg:
+		opts.CARSAKeySize = caRSAKeySize
+	case util.EcAlg:
+		opts.ECSigAlg = util.SupportedECSignatureAlgorithms(ecSigAlg)
+		opts.ECCCurve = util.SupportedEllipticCurves(eccCurve)
+	default:
+		return fmt.Errorf("unknown algorithm type specified (%v)", opts.AlgorithmType)
+	}
+
+	return nil
+}
+
+func specifySelfSignedAlgorithmOptions(opts *util.CertOptions, caRSAKeySize int, algorithmType util.SupportedAlgorithmTypes, ecSigAlg, eccCurve string) error {
+	switch algorithmType {
+	case util.RsaAlg:
+		opts.RSAKeySize = caRSAKeySize
+	case util.EcAlg:
+		opts.ECSigAlg = util.SupportedECSignatureAlgorithms(ecSigAlg)
+		opts.ECCCurve = util.SupportedEllipticCurves(eccCurve)
+	default:
+		return fmt.Errorf("unknown algorithm type specified (%v)", algorithmType)
+	}
+
+	return nil
+}
+
 // NewPluggedCertIstioCAOptions returns a new IstioCAOptions instance using given certificate.
 func NewPluggedCertIstioCAOptions(fileBundle SigningCAFileBundle,
 	defaultCertTTL, maxCertTTL time.Duration, caRSAKeySize int,
+	algorithmType, ecSigAlg, eccCurve string,
 ) (caOpts *IstioCAOptions, err error) {
 	caOpts = &IstioCAOptions{
 		CAType:         pluggedCertCA,
 		DefaultCertTTL: defaultCertTTL,
 		MaxCertTTL:     maxCertTTL,
-		CARSAKeySize:   caRSAKeySize,
+		AlgorithmType:  util.SupportedAlgorithmTypes(algorithmType),
+	}
+
+	err = specifySelfSignedAlgorithmIstioCAOptions(caOpts, caRSAKeySize, ecSigAlg, eccCurve)
```

Comment:
Should we add a log here to check if "ca-cert.pem" used a different signature algorithm then what is specified at startup? Wonder if this could cause some confusion: configuration specifies "ecc", ca-cert has been loaded onto K8s cluster as "rsa". what signing algorithm would be used for DNS cert generation?

Reply:
There are a few other changes I want to put in after this gets merged. I'll put in a log at that time.